engine/chain/internal/bhs/get_merkleroots_from_bhs.go

```diff
@@ -0,0 +1,64 @@
+package bhs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"syscall"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/models"
+)
+
+// GetMerkleRootsFromBHS returns Merkle Roots from Block Header Service
+func (s *Service) GetMerkleRootsFromBHS(ctx context.Context, query url.Values) (*models.MerkleRootsBHSResponse, error) {
+	bhsURL, err := s.createBHSURL("/chain/merkleroot")
+	if err != nil {
+		return nil, err
+	}
+
+	req := s.httpClient.R().
+		SetContext(ctx).
+		SetHeader("Content-Type", "application/json").
+		EnableTrace()
+
+	if s.bhsCfg.AuthToken != "" {
+		req.SetAuthToken(s.bhsCfg.AuthToken)
+	} else {
+		s.logger.Warn().Msg("warning creating Block Headers Service url - auth token is not set. Some requests might not work")
+	}
+
+	var response models.MerkleRootsBHSResponse
+	res, err := req.
+		SetResult(&response).
+		SetQueryParamsFromValues(query).
+		Get(bhsURL.String())
+	if err != nil {
+		if errors.Is(err, syscall.ECONNREFUSED) {
+			return nil, chainerrors.ErrBHSUnreachable.Wrap(err)
+		}
```

Comment:
I did similar thing in my implementations. But slightly different.
I think, we should pick one. (I don't know, for now, which one is better  😄 )

My checking of error:
```golang
if err != nil {
		var e net.Error
		if errors.As(err, &e) {
			return nil, chainerrors.ErrBHSUnreachable.Wrap(err)
		}
		return nil, spverrors.ErrInternal.Wrap(err)
	}
```

Reply:
Hmm, my approach will handle more narrow cases like when server is unresponsive, while yours will also handle cases like timeouts and temporary network failures, so i incline more towards your approach. I will adjust mine